src/lib/memory/numa_memory_resource.hpp

```diff
@@ -0,0 +1,34 @@
+#pragma once
+#include <jemalloc/jemalloc.h>
+#include <unordered_map>
```

Comment:
Nit: I think that the standard format would be
```cpp
#pragma once

#include <unordered_map>

#include <jemalloc/jemalloc.h>
```

I think by using cpplint, this applies to us: https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes